### DISCUSSION TITLE
Reliability of FIFO-based previewer when switching between large files in large code bases

### DISCUSSION THREAD

**Discussion by yingzhu146** [sayanarijit/xplr#244] [Q&A]
Thanks a ton for implementing a previewer, very cool!

One of the things that bothered me about NNN's previewer, however, is that on an average workflow, I have to close and reopen NNN 5-6x because the FIFO becomes unresponsive/overloaded with content and will just show blank. I'm using kitty, but I was wondering why to use a FIFO vs just e.g. use an external kitty pane that is closed and reopened that would also allow full editing in whatever command is used?

Also, maybe separately, what's your workflow for debugging the FIFO workflow, e.g. figuring out what might be causing the preview pane to blank out?

Your work is awesome, as is your engagement with the community so thanks a ton!

Yingzhu

**Comment by sayanarijit** [comment]
Hey thanks for the compliment.
The fifo is nothing but a medium to send the file path to some application or a script that can read it and open it in kitty or some other window.

So for large files, the limitations should be of the window, not the fifo, since the fifo is only passing the file path to a script. However, the workflow shouldn't cause any issue even for large files.

Can you please help me reproduce the issue by giving more info, like how large the files are, which preview script was used etc?

For debugging the fifo, I do `/bin/cat /path/to/fifo` and leave it running in one terminal. Then in another terminal I open xplr and start fifo, close the window opened by the script (rerun /bin/can /parh/to/fifo if required) and navigate. You should see file paths appearing in real-time on the window with the cat command.
